src/testSetup.ts

```diff
@@ -304,6 +305,41 @@ export class TestContext {
     this.configStubs.AuthInfoConfig = { retrieveContents };
   }
 
+  /**
+   * Stub salesforce user authorizations.
+   *
+   * @param users The users to stub.
+   * The key is the username of the admin user and it must be included in the users array in order to obtain the orgId key for the remaining users.
+   * The admin user is excluded from the users array.
+   *
+   */
+  public stubUsers(users: Record<string, MockTestOrgData[]>): Promise<void> {
+    const mockUsers = Object.values(users).flatMap((orgUsers) => orgUsers as unknown as UserFields[]);
```

Comment:
TS is trying to tell you something about these types.

```
Type 'Promise<MockTestOrgData | undefined>' is not assignable to type 'Promise<UserFields | undefined>'.
  Type 'MockTestOrgData | undefined' is not assignable to type 'UserFields | undefined'.
    Type 'MockTestOrgData' is missing the following properties from type 'UserFields': id, lastName, alias, timeZoneSidKey, and 5 more.
```

Should there be an type for UserMock that is a MockTestOrgData but has the additional required props?  Should user mocks be their own type?

How would you use this mock if you don't have these properties (ex: LastName) that aren't on  MockTestOrgData?

They're [in a method](https://github.com/forcedotcom/sfdx-core/blob/f9d69c55fc5261b980e937c06eda6b3cd8a25d6c/src/testSetup.ts#L1057) called `getMockUserInfo` but the typing on that isn't clear either.  

It's used like https://github.com/forcedotcom/sfdx-core/blob/f9d69c55fc5261b980e937c06eda6b3cd8a25d6c/test/unit/org/userTest.ts#L36

I'm with you, where I'd rather give the consumer the option to set props

Reply:
@mshanemc the getMockUserInfo function returns an object with property names that have leading capital letters. Not what I want. I want user fields (UserFields). I'll work to remove the as unknown.